src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,256 @@
+            }
+        }
+        return true;
     }
 
+    @Override
     public boolean addAll(Collection<? extends T> c) {
-        return false;
+        if (c.isEmpty()) {
+            return false;
+        }
+        for (Object o : c) {
+            if (contains(o)) {
+                continue;
+            }
+            resizeIfNeeded();
+            add((T) o);
+        }
+        return true;
     }
 
+    @Override
     public boolean retainAll(Collection<?> c) {
-        return false;
+        boolean changed = false;
+        if (c.isEmpty()) {
+            return false;
+        }
+        for (int i = size - 1; i >= 0; i--) {
+            if (!c.contains(array[i])) {
+                changed = remove(array[i]);
+            }
+        }
+        return changed;
     }
 
+    @Override
     public boolean removeAll(Collection<?> c) {
-        return false;
+        if (c.isEmpty()) {
+            return false;
+        }
+        boolean changed = false;
+        for (Object b : c) {
+            if (remove(b)) {
+                changed = true;
+            }
+        }
+        return changed;
     }
 
+    @Override
     public void clear() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    public static Range<Byte> of(Byte low, Byte high) {
+        Range<Byte> range = new Range<>();
+        if (low.equals(high)) {
+            return range;
+        }
+        for (byte i = low; i <= high; i++) {
+            range.add(i);
+        }
+        return range;
+    }
+
+    public static Range<Short> of(Short low, Short high) {
+        Range<Short> range = new Range<>();
+        if (low.equals(high)) {
+            return range;
+        }
+        for (short i = low; i <= high; i++) {
+            range.add(i);
+        }
+        return range;
+    }
 
+    public static Range<Integer> of(Integer low, Integer high) {
+        Range<Integer> range = new Range<>();
+        if (low.equals(high)) {
+            return range;
+        }
+        for (int i = low; i <= high; i++) {
+            range.add(i);
+        }
+        return range;
+    }
+
+    public static Range<Long> of(Long low, Long high) {
+        Range<Long> range = new Range<>();
+        if (low.equals(high)) {
+            return range;
+        }
+        for (long i = low; i <= high; i++) {
+            range.add(i);
+        }
+        return range;
+    }
+
+    public static Range<Float> of(Float low, Float high) {
+        Range<Float> range = new Range<>();
+        if (low.equals(high)) {
+            return range;
+        }
+        float precision = 0.1f;
+        for (float i = low; i <= high; i += precision) {
+            float number = BigDecimal.valueOf(i).setScale(1, RoundingMode.HALF_UP).floatValue();
+            range.add(number);
+        }
+        return range;
+    }
+
+    public static Range<Double> of(Double low, Double high) {
+        Range<Double> range = new Range<>();
```

Comment:
Use method of(T a, T b, Function<T, T> func). The same for for other **of** methods.